### PR TITLE
test: run axe-playwright a11y tests using test-storybook

### LINKED ISSUE #2384: Add axe testing for our react components

### LINKED ISSUE #2452: investigation: accessibility violations reported by axe
We should go through these stories and determine why we get accessibility violations.

Find the story in Storybook, and open the Accessibility tab to see each violation in context

image

## Reported accessibility violations in these files / stories

Checked tasks have been fixed in #2400 

- [x] packages/react/stories/showcase.stories.tsx  
      story: Showcase
      violation: [Select element must have an accessible name](https://dequeuniversity.com/rules/axe/4.8/select-name?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: Controlled
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: ThousandsOfOptions
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: RemoveAllOptions
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: WithNumberValues
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/Table/Table.stories.tsx  
      story: StickyHeader
      violation: [Scrollable region must have keyboard access](https://dequeuniversity.com/rules/axe/4.8/scrollable-region-focusable?application=axeAPI)
- [x] packages/react/src/components/Table/Table.stories.tsx  
      story: WithFormElements
      violation: [Table header text should not be empty](https://dequeuniversity.com/rules/axe/4.8/empty-table-header?application=axeAPI)
- [x] packages/react/src/components/Table/Table.stories.tsx  
      story: WithFormElements
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
      story: AsLink
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Tooltip/Tooltip.stories.tsx
      story: DefaultOpen
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
      story: Icons
      violation: [Buttons must have discernible text](https://dequeuniversity.com/rules/axe/4.8/button-name?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
      story: IconOnly
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
       story: IconsOnlyPrimary
       violation: [Buttons must have discernible text](https://dequeuniversity.com/rules/axe/4.8/button-name?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
       story: IconsOnlyPrimary
       violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: Floating
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
       story: CustomPlacement
       violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: Status
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: InTabs
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: InButton
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: InButton
      violation: [Buttons must have discernible text](https://dequeuniversity.com/rules/axe/4.8/button-name?application=axeAPI)
- [x] packages/react/src/components/Alert/Alert.stories.tsx  
      story: MedAria
      violation: [ARIA role should be appropriate for the element](https://dequeuniversity.com/rules/axe/4.8/aria-allowed-role?application=axeAPI)
- [x] packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx  
      story: Kontrollert
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Tabs/Tabs.stories.tsx  
      story: Controlled
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/loaders/Skeleton/Skeleton.stories.tsx  
      story: TextExample
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)
- [x] packages/react/src/components/form/Switch/Switch.stories.tsx  
      story: FullWidth
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)
- [x] packages/react/src/components/form/Switch/Switch.stories.tsx  
      story: FullWidthRight
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)

## Violations we will ignore

These violations are examples of invalid usage of components, but these stories only exist to test sizing/alignment, so we don't want to do the work to add correct aria attributes.

- packages/react/stories/testing.stories.tsx  
   story: MediumRow
- packages/react/stories/testing.stories.tsx  
   story: SmallRow
- packages/react/stories/testing.stories.tsx  
   story: LargeRow
- packages/react/stories/testing.stories.tsx  
   story: SmallCol
- packages/react/stories/testing.stories.tsx  
   story: MediumCol
- packages/react/stories/testing.stories.tsx  
   story: LargeCol

### PR DESCRIPTION
**Pull request by unekinn**
ci: enable Biome's "noSvgWithoutTitle" rule, with some exceptions

fix(storybook): delay Storybook tests in Tooltip/DefaultOpen to let tooltip fade in
Fixes false positive color-contrast violation due to axe checking the contrast while the tooltip was not fully faded in yet

refactor(css): animate skeleton background instead of opacity to avoid false positive in a11y contrast test

fix(storybook): disable a11y tests for size/alignment testing stories

fix(react): set Switch wrapper to aria-disabled when disabled=true
Fixes Axe a11y tests reporting insufficient contrast on the description text when the Switch is disabled

fix(storybook): axe violation svg-img-alt in Tabs

fix(storybook): axe violation svg-img-alt in ToggleGroup

fix(storybook): axe violation aria-allowed-role in Alert

fix(storybook): axe violations svg-img-alt & button-name in Badge

fix(storybook): axe violations svg-img-alt & button-name in Button

fix(storybook): axe violations empty-table-header & label in Table

fix(storybook): axe violation scrollable-region-focusable in Table

fix(storybook): axe violation label in Combobox

fix(storybook): axe violation select-name in Showcase

build: make Biome ignore coverage dir by reading .gitignore

ci: add test report publishing also for unit tests

ci: move test coverage report into test.yml to avoid testing twice

test: run axe-playwright a11y tests using test-storybook

Closes #2384 
Closes #2452

### CODE REVIEW

**Comment by unekinn** [comment]
### Feedback necessary

I don't feel like the `checks.yml` workflow is quite the right place for this.

**Suggestion**: I think the Storybook tests should be run after unit tests, which happen in `checks-packages.yml`. However, that particular workflow is currently only run changes to `packages/**`. For the storybook tests to function reliably, we would need to change the workflow config in `checks-packages.yml` to
```yml
  pull_request:
    paths:
      - 'packages/**'
      - 'apps/storybook/**
```

Is that okay?

**Comment by unekinn** [comment]
### Feedback necessary

@mimarz I don't feel like the `checks.yml` workflow is quite the right place for this to run.

**Suggestion**: I think the Storybook tests should be run after unit tests, which happen in `checks-packages.yml`. A bonus is that `yarn build` is already run there, which saves some build time. However, that particular workflow is currently only run on changes to `packages/**`.

For the storybook tests to function reliably, we would need to change the workflow config in `checks-packages.yml` to
```yml
  pull_request:
    paths:
      - 'packages/**'
      - 'apps/storybook/**
```

Is that okay?

**Comment by mimarz** [comment]
> ### Feedback necessary
> 
> 
> 
> @mimarz I don't feel like the `checks.yml` workflow is quite the right place for this to run.
> 
> 
> 
> **Suggestion**: I think the Storybook tests should be run after unit tests, which happen in `checks-packages.yml`. A bonus is that `yarn build` is already run there, which saves some build time. However, that particular workflow is currently only run on changes to `packages/**`.
> 
> 
> 
> For the storybook tests to function reliably, we would need to change the workflow config in `checks-packages.yml` to
> 
> ```yml
> 
>   pull_request:
> 
>     paths:
> 
>       - 'packages/**'
> 
>       - 'apps/storybook/**
> 
> ```
> 
> 
> 
> Is that okay?

Jepp. It could probably be renamed to a "Test" action as the checks changed a bit after we moved to biomejs.

File: biome.jsonc [hunk 1]
```diff
@@ -1,17 +1,18 @@
 {
   "$schema": "https://biomejs.dev/schemas/1.8.3/schema.json",
   "organizeImports": { "enabled": true },
+  "vcs": {
+    "enabled": true,
+    "clientKind": "git",
+    "useIgnoreFile": true
+  },
```
Comment (mimarz) [inline biome.jsonc:8]: Oh neat! Didn't notice this feature from biome before <3

**Review by mimarz — APPROVED** [review]
Nice work! 

Interesting to see how this testing will go :D

**Pull request merged**